### ISSUE TITLE
Extensible list of enrichable content

### ISSUE THREAD

**Issue by georgecrawford** [Financial-Times/spoor-enrichment#105]
@commuterjoy - what do you think about this?

I remember liking the `Add-ons` feature for Keen.io (https://keen.io/docs/api/#url-parser). It allows the author of a tracking call to specify which of the keys should be enriched into a parsed URL, Geo coordinates, parsed UA, etc.

I've found that there are a number of different keys in the JSON I'm sending where I might want to have a URL deconstructed, and it would be great to be able to specify those declaratively rather than conforming to a predetermined list.

Perhaps this is overkill, and I should just raise a PR here if I want more keys enriched. Don't know really. Just wanted to suggest a more flexible approach.

**Comment by commuterjoy** [comment]
Yup, nice idea. We could even just auto-expand anything that looks like a URL?

**Comment by georgecrawford** [comment]
It's just that we might make use of this in the future for types which aren't so easily recognisable. I like the declarative approach of opting-in to add-ons.

**Comment by commuterjoy** [comment]
Yes, fair point. I think the more diverse the events being sent to spoor the more something like this will be useful. Or failing that, running different pipelines for different events.
